SNOW-1296059: Exclude cloud providers from driver, only required for compile

sfc-gh-jmartinezramirez opened the pull request:
### Description
Exclude cloud providers from driver, only required for compile
Move logic to Snowflake.Data.Core to only have compile dependencies to cloud providers. 
Add projects for dependencies nuget generation for each cloud provider.

### Checklist
- [x] Code compiles correctly
- [x] Code is formatted according to [Coding Conventions](../blob/master/CodingConventions.md)
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing (`dotnet test`)
- [x] Extended the README / documentation, if necessary
- [x] Provide JIRA issue id (if possible) or GitHub issue id in PR name

sfc-gh-mhofman commented on Snowflake.Data.Common/Snowflake.Data.Common.csproj:
```diff
@@ -0,0 +1,78 @@
+  <PropertyGroup>
+    <TargetFrameworks>net6.0;net471;net472</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">net6.0</TargetFrameworks>
+    <Title>Snowflake.Data.Common</Title>
```
It seems to me that "common" has some meaning related to the underlying tools or utilities common for some other solution.
My thinking is we should name it as Snowflake.Data.Core?
WDYT? @sfc-gh-knozderko  @sfc-gh-jmartinez @sfc-gh-erojaslizano

sfc-gh-knozderko replied:
Core has the problem that it can suggest it to be for .NET core and not .net framework.

sfc-gh-jmartinez replied:
@sfc-gh-mhofman We have Snowflake.Data.Core was previously the first option, and the concern of a misunderstanding that the package is only for .Net Core made us change the name, but maybe it is not a problem. I agree to use Snowflake.Data.Core

sfc-gh-mhofman commented on Snowflake.Data/Snowflake.Data.csproj:
```diff
@@ -1,68 +1,31 @@
+        <Description>Snowflake Connector for .NET</Description>
+        <Company>Snowflake Computing, Inc</Company>
+        <Product>Snowflake Connector for .NET</Product>
+        <Authors>howryu, tchen</Authors>
```
we could change here to Snowflake

sfc-gh-jmartinez replied:
Which line are you sugesting to change Company, authors? @sfc-gh-mhofman

sfc-gh-mhofman reviewed (commented):
This looks promising @sfc-gh-sthyagaraj 
We end up with
1. current thick library: Snowflake.Data (with GCP,AZURE and AWS out-of-the-box)
2. thin library: Snowflake.Data.Common (yet my suggestion would be to think and maybe change to e.g. Snowflake.Data.Core)
3. cloud specific extensions up to customer needs: 
- Snowflake.Data.AWS 
- Snowflake.Data.Azure
- Snowflake.Data.GCP

sfc-gh-knozderko commented on Snowflake.Data/Snowflake.Data.csproj:
```diff
@@ -1,68 +1,31 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
```
The namespace is the same.

sfc-gh-jmartinezramirez committed:
Update readme

sfc-gh-jmartinezramirez committed:
Sync csproj properties with latest version

sfc-gh-jmartinez commented on Snowflake.Data.Azure/Snowflake.Data.Azure.csproj:
```diff
@@ -0,0 +1,25 @@
+    </PropertyGroup>
+
+    <ItemGroup>
+      <PackageReference Include="Azure.Storage.Blobs" Version="12.13.0" />
```
Added explicit dependency.

sfc-gh-jmartinezramirez committed:
Added explicit Azure.Storage.Common dependency.
Set alpha version to projects 4.0.1-alpha01.
Removed not required set of version, it is calculated base on the Version property.

sfc-gh-jmartinezramirez committed:
Disable publish of new splitted dependencies in deploy.bat

sfc-gh-jmartinezramirez committed:
Exclude cloud providers from the new Snowflake.Data.Core.
Move logic to Snowflake.Data.Core to only have compile dependencies to cloud providers.
Add projects for dependencies nuget generation for each cloud provider.
Update deploy bat to include new packages

sfc-gh-mcepiga commented:
After years of inactivity, I'm closing this as difficult to salvage. The idea itself stays in consideration.
